integration-tests/build.gradle

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.nio.file.Paths
+
+plugins {
+    id('idea')
+    // A Java library
+    id('java-library')
+    // which is published
+    id('maven-publish')
+}
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            from components.java
+            groupId project.group
+            artifactId "${archivesBaseName}"
+            version System.getenv("CODE_VERSION") ?: "${version}"
+        }
+    }
+}
+
+def integrationMaxHeapSize = System.getenv("INTEGRATION_MAX_HEAP_SIZE") ?: "3000M"
+println("Using ${integrationMaxHeapSize} maxHeapSize")
+
+def integrationMaxParallelForks = (System.getenv("INTEGRATION_MAX_PARALLEL_FORKS") ?: "4") as int
+println("Using ${integrationMaxParallelForks} maxParallelForks")
+
+def integrationEnableMtls = (System.getenv("INTEGRATION_MTLS_ENABLED") ?: "true") as boolean
+println("Using mTLS for tests? ${integrationEnableMtls}")
+
+configurations {
+    // remove netty-all dependency coming from spark
+    all*.exclude(group: 'io.netty', module: 'netty-all')
+    configureEach {
+        resolutionStrategy {
+            // Force set the vertx versions that do not depend on a newer jackson version
+            force("io.vertx:vertx-web:${vertxVersion}")
+            force("io.vertx:vertx-core:${vertxVersion}")
+            force("io.vertx:vertx-auth-common:${vertxVersion}")
+            force("io.vertx:vertx-dropwizard-metrics:${vertxVersion}")
+            force("io.vertx:vertx-web-client:${vertxVersion}")
+        }
+    }
+}
+
+dependencies {
+    implementation("io.vertx:vertx-web:${project.vertxVersion}") {
+        exclude group: 'junit', module: 'junit'
+    }
+    implementation("io.vertx:vertx-dropwizard-metrics:${project.vertxVersion}")
+    implementation("io.vertx:vertx-web-client:${project.vertxVersion}")
+
+    testImplementation(group: 'net.java.dev.jna', name: 'jna', version: '5.9.0')
+    implementation(project(path: ":integration-framework"))
+    testImplementation('com.google.inject:guice:4.2.2')
+    testRuntimeOnly 'com.fasterxml.jackson.core:jackson-annotations:2.14.2'
+}
+
+test {
```

Comment:
I think the tests in `integration-tests` end up running in `unit_java8` and `unit_java11`. It would be undesired. 
Please change to integrationTest as defined in `server/build.gradle`. 

```
def integrationTest = task("integrationTest")

['integrationTestLightWeight', 'integrationTestHeavyWeight'].each { name ->
...
```